src/coreComponents/python/modules/pygeosx_tools_package/pygeosx_tools/mesh_interpolation.py

```diff
@@ -0,0 +1,202 @@
+    """
+    nodes = []
+    with open(fname, 'r') as f:
+        for line in f:
+            if ('#' not in line):
+                tmp = line.split(',')
+                if ('.' in line):
+                    nodes.append([float(x) for x in tmp])
+
+    # Convert values
+    nodes = np.array(nodes)
+    x = np.ascontiguousarray(np.squeeze(nodes[:, 2])) / 3.281
+    y = np.ascontiguousarray(np.squeeze(nodes[:, 0])) / 3.281
+    z = (np.ascontiguousarray(np.squeeze(nodes[:, 1])) - z_origin) / -3.281
+
+    # Generate four interpolators to reconstruct the surface
+    interp_a = interpolate.LinearNDInterpolator((x, y), z)
+    interp_b = interpolate.NearestNDInterpolator((x, y), z)
+    interp_c = interpolate.NearestNDInterpolator((x, y), x)
+    interp_d = interpolate.NearestNDInterpolator((x, y), y)
+
+    return interp_a, interp_b, interp_c, interp_d
+
+
+def build_conforming_tables(property_interps, surface_interps, axes, extrapolate_surfaces=True):
+    """
+    Build a set of conforming tables property, surface interpolators
+
+    Args:
+        property_interps a dict of 1D property interpolators with depth
+        surface_interps a dict of interpolators for surfaces
+        axes a list of x,y,z axis definitions
+        extrapolate_surfaces a flag to indicate whether surface elevations should be extrapolated (default=True)
+
+    Returns:
+        tuple: Conforming tables, surface elevations at grid, grid values
+    """
+    # Sort the surfaces by depth at the origin
+    surfaces = list(surface_interps.keys())
+    z_pilot = [surface_interps[ka][1](0.0, 0.0) for ka in surfaces]
+    Ia = np.argsort(z_pilot)
+    z_pilot = np.array(z_pilot)[Ia]
+    surfaces = np.array(surfaces)[Ia]
+
+    # Interpolate values onto a new grid
+    x_grid, y_grid, z_grid = axes
+    zo = np.concatenate([[-1e6], z_pilot, [0.0]], axis=0)
+    X, Y, Z = np.meshgrid(x_grid, y_grid, z_grid, indexing='ij')
+    conforming_tables = {k: np.zeros(np.shape(X)) for k in property_interps.keys()}
+    surface_elevations = np.zeros((len(x_grid), len(y_grid), len(surfaces)))
+
+    for ii in range(len(x_grid)):
+        for jj in range(len(y_grid)):
+            for kk, s in zip(range(len(surfaces)), surfaces):
+                # Find the current layer positions
+                tmp_z = surface_interps[s][0](x_grid[ii], y_grid[jj])
+                if np.isnan(tmp_z):
+                    # The point is out of the surface xy range
+                    tmp_z = surface_interps[s][1](x_grid[ii], y_grid[jj])
+                    if extrapolate_surfaces:
+                        # Find nearest point, add in the average delta_z from other layers
+                        nearest_x = surface_interps[s][2](x_grid[ii], y_grid[jj])
+                        nearest_y = surface_interps[s][3](x_grid[ii], y_grid[jj])
+                        other_delta_z = []
+                        for kk in range(0, len(surfaces)):
```

Comment:
Do you really want to shadowe previous `kk`?

Reply:
This function was specific to a problem I was working on and probably shouldn't belong in the repository